### PR TITLE
Use TRIM function to remove whitespace in SQL names

### PR DESCRIPTION
**Pull request by qpresley**
Many of the discovery functions were pulling contents of columns and ignoring whitespace.  DB2 stores column names, table names, schema names, etc. with white space so using the TRIM function is a safe way to avoid bad comparisons when searching for names in the catalog tables.

### CODE REVIEW

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by bajtos** [comment]
@superkhau @Amir-61 could one of you please review this pull request for @qpresley?

**Comment by qpresley** [comment]
@slnode please test

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

**Comment by qpresley** [comment]
@slnode test please

File: lib/discovery.js [hunk 1]
```diff
@@ -63,20 +63,25 @@ function mixinDiscovery(DB2, db2) {
 
     if (options.all && !schema) {
       sqlTables = paginateSQL('SELECT \'table\' AS "type",' +
-        ' tabname AS "name", tabschema AS "owner", property as "property"' +
+        ' TRIM(tabname) AS "name",' +
+        ' TRIM(tabschema) AS "owner",' +
+        ' TRIM(property) as "property"' +
```
Comment (superkhau) [inline lib/discovery.js]: Any reason why sometimes `AS` is uppercase and the other ones are lowercase `as`?

File: lib/discovery.js [hunk 2]
```diff
@@ -211,25 +216,23 @@ function mixinDiscovery(DB2, db2) {
...
     if (schema) {
-      sql = paginateSQL('SELECT tabschema AS "owner",' +
-        ' tabname AS "tableName",' +
-        ' colname AS "columnName",' +
+      sql = paginateSQL('SELECT TRIM(tabschema) AS "owner",' +
+        ' TRIM(tabname) AS "tableName",' +
+        ' TRIM(colname) AS "columnName",' +
         ' typename AS "dataType",' +
         ' length AS "dataLength",' +
-        // ' numeric_precision AS "dataPrecision",' +
```
Comment (superkhau) [inline lib/discovery.js]: Is this deletion intentional? If so, should be in a seperate commit since it's unrelated to trim changes.

File: package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
   "scripts": {
     "pretest": "eslint ./ && jscs ./",
     "lint": "eslint ./ && jscs ./",
-    "test": "mocha --timeout 10000"
+    "test": "mocha --timeout 15000"
```
Comment (superkhau) [inline package.json]: This should also be in a separate commit as it's unrelated to the TRIM changes.

**Comment by superkhau** [comment]
@qpresley Just minor nitpicks, but otherwise LGTM. I'll leave it up to you as to whether to address them or not.

File: lib/discovery.js [hunk 4]
```diff
@@ -63,20 +63,25 @@ function mixinDiscovery(DB2, db2) {
...
+        ' TRIM(tabschema) AS "owner",' +
+        ' TRIM(property) as "property"' +
         ' FROM syscat.tables where substr(property,20,1) NOT LIKE \'Y\'',
         'table_schema, table_name', options);
     } else if (schema) {
       sqlTables = paginateSQL('SELECT \'table\' AS "type",' +
-        ' tabname AS "name", tabschema AS "schema", property as "property"' +
+        ' TRIM(tabname) AS "name",' +
+        ' TRIM(tabschema) AS "schema",' +
+        ' TRIM(property) as "property"' +
```
Comment (superkhau) [inline lib/discovery.js]: ditto

File: lib/discovery.js [hunk 5]
```diff
@@ -63,20 +63,25 @@ function mixinDiscovery(DB2, db2) {
...
         ' WHERE tabschema=\'' + schema + '\' AND' +
         ' SUBSTR(property, 20, 1) NOT LIKE \'Y\'',
         'table_schema, table_name', options);
     } else {
       sqlTables = paginateSQL('SELECT \'table\' AS "type",' +
-        ' tabname AS "name", ' +
-        ' tabschema AS "owner", property as "property" FROM syscat.tables' +
+        ' TRIM(tabname) AS "name", ' +
+        ' TRIM(tabschema) AS "owner",' +
+        ' TRIM(property) as "property" FROM syscat.tables' +
```
Comment (superkhau) [inline lib/discovery.js]: ditto

**Comment by qpresley** [comment]
Thanks @superkhau much of what you caught has been there for a while .. and unfortunately the package.json update is related in that these are all changes being made to get CI functioning reliably.  There is no good reason for AS and as other than I likely copied and pasted it in.  The commented out line I thought I deleted.  I may fix those up before merging since it looks like I still have some CI infrastructure updates to make to get this to pass.

**Comment by qpresley** [comment]
@slnode test please
